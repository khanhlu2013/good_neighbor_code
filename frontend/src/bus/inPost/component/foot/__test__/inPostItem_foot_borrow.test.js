import React from "react";
import { shallow } from "enzyme";
import InPostItemFootBorrow from "../inPostItem_foot_borrow";
import LoadingIcon from "../../../../../share/loadingIcon";

describe("inPostItem_foot_borrow", () => {
  describe("loading icon", () => {
    it("is display during aware approve", () => {
      const onAwareApprovePost = jest.fn();
      const onReturnPost = jest.fn();

      const wrap = shallow(
        <InPostItemFootBorrow
          myBorrowShareId={"myBorrowShareIdStub"}
          isAwareApproveBorrowShare={false}
          isAwaringShare={true}
          isReturningShare={false}
          onAwareApprovePost={onAwareApprovePost}
          onReturnPost={onReturnPost}
        />
      );
      expect(wrap.find(LoadingIcon).props().text).toBe("aware approve");
      expect(wrap).toMatchSnapshot();
    });

    it("is display during return", () => {
      const onAwareApprovePost = jest.fn();
      const onReturnPost = jest.fn();

      const wrap = shallow(
        <InPostItemFootBorrow
          myBorrowShareId={"myBorrowShareIdStub"}
          isAwareApproveBorrowShare={true}
          isAwaringShare={false}
          isReturningShare={true}
          onAwareApprovePost={onAwareApprovePost}
          onReturnPost={onReturnPost}
        />
      );
      expect(wrap.find(LoadingIcon).props().text).toBe("return");
      expect(wrap).toMatchSnapshot();
    });
  });

  it("trigger onAwareApprovePost and onReturnPost callback correctly", () => {
    const myBorrowShareId = "myBorrowShareIdStub";
    const onAwareApprovePost = jest.fn();
    const onReturnPost = jest.fn();

    const wrap = shallow(
      <InPostItemFootBorrow
        myBorrowShareId={myBorrowShareId}
        isAwareApproveBorrowShare={false}
        isAwaringShare={false}
        isReturningShare={false}
        onAwareApprovePost={onAwareApprovePost}
        onReturnPost={onReturnPost}
      />
    );
    wrap.find("#outPostItem-awareApproveBtn-react").simulate("click");
    expect(onAwareApprovePost.mock.calls).toHaveLength(1);
    expect(onAwareApprovePost.mock.calls[0][0]).toBe(myBorrowShareId);

    wrap.find("#outPostItem-returnBtn-react").simulate("click");
    expect(onReturnPost.mock.calls).toHaveLength(1);
    expect(onReturnPost.mock.calls[0][0]).toBe(myBorrowShareId);
  });

  it("match snapshot", () => {
    const onAwareApprovePost = jest.fn();
    const onReturnPost = jest.fn();

    //not yet aware share
    let wrap = shallow(
      <InPostItemFootBorrow
        myBorrowShareId={"myBorrowShareIdStub"}
        isAwareApproveBorrowShare={false}
        isAwaringShare={false}
        isReturningShare={false}
        onAwareApprovePost={onAwareApprovePost}
        onReturnPost={onReturnPost}
      />
    );
    expect(wrap).toMatchSnapshot("not yet aware share");

    //already awared share
    wrap = shallow(
      <InPostItemFootBorrow
        myBorrowShareId={"myBorrowShareIdStub"}
        isAwareApproveBorrowShare={true}
        isAwaringShare={false}
        isReturningShare={false}
        onAwareApprovePost={onAwareApprovePost}
        onReturnPost={onReturnPost}
      />
    );
    expect(wrap).toMatchSnapshot("already awared share");
  });
});
