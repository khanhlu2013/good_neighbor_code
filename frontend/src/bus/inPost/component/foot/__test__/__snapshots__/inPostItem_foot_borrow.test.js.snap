// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inPostItem_foot_borrow loading icon is display during aware approve 1`] = `
<div
  className="text-success"
>
  request approved.
  <span
    className="ml-1"
  >
    <LoadingIcon
      text="aware approve"
    />
  </span>
  <button
    className="btn btn-sm btn-warning ml-1"
    id="outPostItem-returnBtn-react"
    onClick={[Function]}
  >
    return item
  </button>
</div>
`;

exports[`inPostItem_foot_borrow loading icon is display during return 1`] = `
<div
  className="text-success"
>
  request approved.
  <span
    className="ml-1"
  >
    <span>
      You've received.
    </span>
  </span>
  <LoadingIcon
    text="return"
  />
</div>
`;

exports[`inPostItem_foot_borrow match snapshot: already awared share 1`] = `
<div
  className="text-success"
>
  request approved.
  <span
    className="ml-1"
  >
    <span>
      You've received.
    </span>
  </span>
  <button
    className="btn btn-sm btn-warning ml-1"
    id="outPostItem-returnBtn-react"
    onClick={[Function]}
  >
    return item
  </button>
</div>
`;

exports[`inPostItem_foot_borrow match snapshot: not yet aware share 1`] = `
<div
  className="text-success"
>
  request approved.
  <span
    className="ml-1"
  >
    <button
      className="btn btn-sm btn-success"
      id="outPostItem-awareApproveBtn-react"
      onClick={[Function]}
    >
      confirm approved
    </button>
  </span>
  <button
    className="btn btn-sm btn-warning ml-1"
    id="outPostItem-returnBtn-react"
    onClick={[Function]}
  >
    return item
  </button>
</div>
`;
